test/plugin/test_out_google_cloud_grpc.rb

```diff
@@ -224,12 +224,22 @@ def api_client
 
   # GRPC logging mock that successfully logs the records.
   class GRPCLoggingMockService < Google::Logging::V1::LoggingService::Service
-    def initialize(requests_received)
+    def initialize(stub_params, requests_received)
       super()
       @requests_received = requests_received
+      @expected_requests = stub_params.clone
     end
 
     def write_log_entries(request, _call)
+      matched = @expected_requests.take_while do |expected|
+        "projects/#{expected[:project_id]}/logs/#{expected[:log_name]}" \
+          == request.log_name
```

Comment:
I meant that you could construct a list of expected string values for `request.log_name`. Something like:
```
@expected_log_names = stub_params.map { |r| "projects/#{r[:project_id]}/logs/#{r[:log_name]}" }
```
and then iterate over that:
```
matched = @expected_log_names.take_while do |expected|
  expected == request.log_name
end
```
WDYT?